pkg/trace/info/stats.go

```diff
@@ -288,6 +284,49 @@ func (s *SpansMalformed) String() string {
 	return mapToString(s.tagValues())
 }
 
+const (
+	maxAbsSamplingPriority = 10
+	numPriorityCounters    = maxAbsSamplingPriority*2 + 1
+)
+
+// samplingPriorityStats holds the sampling priority metrics that will be reported every 10s by the agent.
+type samplingPriorityStats struct {
```

Comment:
I think that the implementation could be a lot more elegant if you added a constructor:
```go
// newSamplingPriorityStats returns a new samplingPriorityStats which is able to counter
// priorities in the range [-maxAbsPriority, maxAbsPriority].
func newSamplingPriorityStats(maxAbsPriority int) *samplingPriorityStats {
    n := maxAbsPriority*2 + 1
    return &samplingPriorityStats{
        counts: make([]int64, n),
        max: maxAbsPriority,
    }
}
```
This will rid you of the constants (allowing you to loop over the `counts` using `len(s.counts)`) and make everything more succint.

What do you think? Would it work?

Reply:
Hi Gabriel,

I see your point, but introducing a "dynamic" size for the array (as in, it can defined at runtime during creation), requires handling additional complexity. Specifically the update function should handle combination of arrays of different sizes, the constructor should verify that the given max priority is valid and the user of this struct (the Stats code) would have to call this constructor but it doesn't have a constructor itself. 

However, I can make use of the len(s.counts) all the same and remove one constant, leaving only one to control the desired maxAbsPriority. Would that be acceptable?

Reply:
Fair points. Yes I think that's totally acceptable.